vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -401,6 +403,13 @@ private void sendFile(RoutingContext context, String file, FileProps fileProps)
           // guess content type
           String contentType = MimeMapping.getMimeTypeForFilename(file);
           if (contentType != null) {
+            int li = file.lastIndexOf(46);
```

Comment:
For clarity, can you move the logic to determine the file extension to a private method?

When it's done, change the code to guess the content type with `MimeMapping#getMimeTypeForExtension` instead of `MimeMapping#getMimeTypeForFilename`. That will avoid computing the file extension twice.